File: solutions/java/roman-numerals/1/src/main/java/RomanNumerals.java
public class RomanNumerals {

    private static final int[] ARABIC_NUMERALS = new int[]{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};

    private static final String[] ROMAN_NUMERALS = new String[]{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};

    private final int number;

    public RomanNumerals(int number) {
        this.number = number;
    }

    public String getRomanNumeral() {
        String romanNumeral = "";
        int n = number;
        for (int i = 0; i < ARABIC_NUMERALS.length; i++) {
            while (n >= ARABIC_NUMERALS[i]) {
                romanNumeral += ROMAN_NUMERALS[i];
                n -= ARABIC_NUMERALS[i];
            }
        }
        return romanNumeral;
    }
}

File: solutions/kotlin/roman-numerals/1/src/main/kotlin/RomanNumerals.kt
object RomanNumerals {

    private val ARABIC_NUMERALS = intArrayOf(1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1)

    private val ROMAN_NUMERALS = arrayOf("M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I")


    fun value(n: Int): String {
        var num = n;
        val sb = StringBuilder()
        ARABIC_NUMERALS.forEachIndexed { i, x ->
            run {
                while (num >= x) {
                    sb.append(ROMAN_NUMERALS[i])
                    num -= x
                }
            }
        }
        return sb.toString()
    }
}
